### DISCUSSION TITLE
Python incompatibility issue when building gem5 v20.1.0.4 for SPEC CPU 2017

### DISCUSSION THREAD

**Discussion by m-ert** [gem5/gem5#1806] [Q&A] [answered]
I am new and Im following the [SPEC tutorial](https://www.gem5.org/documentation/gem5art/tutorials/spec-tutorial) which requires **gem5 v20.1.0.4** and Im getting Python version related errors when trying to build via `scons build/X86/gem5.opt -j8`.

On my Ubuntu 24.04 system, I had python 3.12 by default and got the following error saying there are deprecated and removed stuff such as imp module:
```
...
line 285:
    exec(compile(scriptdata, scriptname, 'exec'), call_stack[-1].globals)
  File "/home/mhsn/gem5-resources/src/spec-2017/gem5/build/X86/SConscript", line 45:
    import imp
```

Tried building with Python `3.12`, `3.10`, `3.8`, `3.4` using **virtual environment** but all of them gives errors.

Errors look all similar and seem to be some python version incompatibility.

gem5 v24.0.0.1 builds and works fine. I could not find the documentation for running gem5 20.1.0.4. Do I need to downgrade Ubuntu?

**Comment by powerjg** [comment, marked as answer]
gem5 20.1 is no longer supported. I would suggest trying to use 24.0 with the tutorial. In fact. gem5art isn't well supported anymore either. It has been migrated to gem5-resources. This page may be more helpful: https://github.com/gem5/gem5-resources/tree/stable/src/spec-2017

**Reply by m-ert** [reply]
Thank you Mr. Lowe-Power.
